src/sentry/migrations/0384_backfill_monitor_env_checkins.py

```diff
@@ -0,0 +1,81 @@
+# Generated by Django 2.2.28 on 2023-03-14 23:45
+
+from django.db import connection, migrations
+from psycopg2.extras import execute_values
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+BATCH_SIZE = 100
+DEFAULT_ENVIRONMENT_NAME = "production"
+
+UPDATE_QUERY = """
+    UPDATE sentry_monitorcheckin
+    SET monitor_environment_id = data.monitor_environment_id
+    FROM (VALUES %s) AS data (id, monitor_environment_id)
+    WHERE sentry_monitorcheckin.id = data.id"""
+
+
+def backfill_monitor_checkins(apps, schema_editor):
+    MonitorCheckIn = apps.get_model("sentry", "MonitorCheckIn")
+    MonitorEnvironment = apps.get_model("sentry", "MonitorEnvironment")
+
+    queryset = RangeQuerySetWrapperWithProgressBar(
+        MonitorCheckIn.objects.all().values_list(
+            "id",
+            "project_id",
+            "monitor",
+            "monitor_environment",
+        ),
+        result_value_getter=lambda item: item[0],
+    )
+
+    cursor = connection.cursor()
+    batch = []
+    for monitor_checkin_id, project_id, monitor_id, monitor_environment_id in queryset:
+        if monitor_environment_id:
+            continue
+
+        try:
+            monitor_environment = MonitorEnvironment.objects.filter(monitor_id=monitor_id)[0]
+        except IndexError:
```

Comment:
this shouldn't happen but I'd like to check redash to confirm we don't have any naked `Monitor` with no associated `MonitorEnv`